### PR TITLE
Update client address to be consistent with server address

### LINKED ISSUE #251: Should domain name be allowed in client.address?
Either we need allow a domain name in the client.address as well, or we forbid it in the server address.
(I think mixing name and address is confusing and means potential information loss, as we can no longer report the name + address of the logical client/server at the same time, since both are stored in "address", but the HTTP conventions rely on this quirk already, and are in "Feature Freeze" -- you may call a potential split a bugfix though?)

_Originally posted by @Oberon00 in https://github.com/open-telemetry/semantic-conventions/pull/244#discussion_r1289799568_

### PR DESCRIPTION
**Pull request by trask**
Resolves #251
Closes #290 

## Changes

Updates `client.address` to allow domain names for consistency with `server.address`.

This is consistent with the ECS definition of [`client.address`](https://www.elastic.co/guide/en/ecs/current/ecs-client.html#field-client-address).

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] [CHANGELOG.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CHANGELOG.md) updated for non-trivial changes.
* [x] ~[schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.~

### CODE REVIEW

File: docs/http/http-spans.md [hunk 1]
```diff
@@ -334,7 +334,7 @@ If the route cannot be determined, the `name` attribute MUST be set as defined i
 | Attribute  | Type | Description  | Examples  | Requirement Level |
 |---|---|---|---|---|
 | `http.route` | string | The matched route (path template in the format used by the respective server framework). See note below [1] | `/users/:userID?`; `{controller}/{action}/{id?}` | Conditionally Required: If and only if it's available |
-| [`client.address`](../general/attributes.md) | string | Client address - IP address or Unix domain socket name. [2] | `83.164.160.102` | Recommended |
+| [`client.address`](../general/attributes.md) | string | Client address - domain name if available without reverse DNS lookup, otherwise IP address or Unix domain socket name. [2] | `83.164.160.102` | Recommended |
```
Comment (lmolkova) [inline docs/http/http-spans.md:337]: I really like this change - e.g. the `for` directive in [forwarded](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded#directives) header could be an obfuscated id  which not IP address.

**Review by lmolkova — APPROVED** [review]

**Review by AlexanderWert — APPROVED** [review]

**Review by Oberon00 — APPROVED** [review]
Makes sense. We should apply the same to source and destination conventions (and remove source.domain, destination.domain).
Please also close #290 from the PR description for clarity.
(EDIT: I tried to edit it in but it seems you can now only close issues but not PRs from PRs. In that case please close #290 manually after merging this one)

**Comment by Oberon00** [comment]
(Addendum: I tried to edit the "closes #290" in but it seems you can now only close issues but not PRs from PRs. In that case please close #290 manually after merging this one))

**Review by arminru — APPROVED** [review]

**Comment by arminru** [comment]
> (Addendum: I tried to edit the "closes #290" in but it seems you can now only close issues but not PRs from PRs. In that case please close #290 manually after merging this one)

@Oberon00 I added "Closes #290" again in the description before merging and it seems to have worked: https://github.com/open-telemetry/semantic-conventions/pull/290#event-10348821312

**Comment by Oberon00** [comment]
Nice! For some reason it is not linked in the right sidebar

**Pull request merged**